### DISCUSSION TITLE
ErrorException Trying to access array offset on value of type int

### DISCUSSION THREAD

**Discussion by austinfiddes** [laravel/framework#36176] [Q&A]
Hello all, 

I am brand new to Laravel and recently have been following along with a tutorial on learning how to work with it, and in this tutorial we learn how to create a to do list. I am at the very end of this tutorial and at this point we are using livewire to allow us to create new to do step fields in addition to already existing ones when editing a to do. When I go to create a new to do step I get this error: 

![error-message](https://user-images.githubusercontent.com/39268837/107156783-9180f000-693d-11eb-9dc4-84e57aa2eb9a.png)

I feel the error message gives a pretty clear and obvious statement, but after lots of searching I cannot find a solution. 

It tells me line 8 of this edit-step.blade.php is where my error is coming from. Does anyone have any ideas for a solution to this?

![code](https://user-images.githubusercontent.com/39268837/107156905-3dc2d680-693e-11eb-9b9b-5cd061b6e095.png)

All help is greatly appreciated

**Comment by Muhammad-Sarfaraz** [comment]
You are probably trying accessing a non-array type,or It's a PHP versioning issue please have a look [#30737](https://github.com/laravel/framework/issues/30737)

**Reply by austinfiddes** [reply]
Thanks for the reply Muhammed, during my previous efforts to try and solve this issue I did stumble upon this one you've linked me, but I wasn't able to make much use of it. If more info helps I also have an EditSteps controller, in there I have a mount function where I convert $steps to array using $this->steps = $steps->toArray();, also my versions are PHP(v 8.0.1) and Laravel(v 7.30.4)

**Reply by austrowGC** [reply]
If you `var_dump` `$steps` and/or `$step`, what does the output say the type is?

**Reply by Muhammad-Sarfaraz** [reply]
@austinfiddes It happens while you try to get the index 'arrayIndex' of an array that does not exist.I think you should try [dd()](https://laravel.com/docs/8.x/collections#method-dd) or ddd() and have look on your **$steps** variable,or you can post it here,it will be helpful, and for further please use [@forelse](https://laravel.com/docs/8.x/blade#loops).

**Reply by austinfiddes** [reply]
@austrowGC @Muhammad-Sarfaraz I did try doing a dd on my $steps, I will attach the results below. 

Here I did a dd($this->steps)
![dd-this-steps](https://user-images.githubusercontent.com/39268837/107397954-61ac2680-6abc-11eb-944b-eb5d26d3904f.png)

Here I did a dd($steps)
![dd-steps](https://user-images.githubusercontent.com/39268837/107398030-78527d80-6abc-11eb-95c5-303b30be73f5.png)

And here is my EditStep controller if this would be helpful to see.
![edit-step](https://user-images.githubusercontent.com/39268837/107398094-886a5d00-6abc-11eb-81f4-a8d7306b8c75.png)

**Reply by Muhammad-Sarfaraz** [reply]
@austinfiddes can you please  use dd() also in your view,I think your data is not rendered,in your views.

**Reply by austinfiddes** [reply]
@Muhammad-Sarfaraz Here is the result from doing a dd($steps) within the foreach of the view
![dd-steps-view](https://user-images.githubusercontent.com/39268837/107427111-48b46d00-6ade-11eb-8a04-fbf51fd506cc.png)

**Comment by Muhammad-Sarfaraz** [comment]
Try Follow these steps, clear your view **php artisan view:clear** and **php artisan cache:clear** and remove your html code inside foreach loop and try only echo   **{{$step['name']}}** inside foreach loop,It will helps you to identify which portion of your code is causing this errors.

**Comment by iamjayrajsinh** [comment]
Bro i have been facing the same problem did you have any solution for it..???

**Reply by prudentjag** [reply]
in your controller when you are fetching the data use `$result = DB::table('your table')->where('condition')->get()` because I was using `$result = DB::table('your table')->where('condition')->first()` and it was giving me that error, the in blade you can use forelse to loop it

**Reply by jyrjsinhatfolium** [reply]
i think you misunderstood the problem but i have got solution for my problem thanks for your approach @prudentjag..

**Comment by raidski20** [comment]
I faced the same issue, and after some debugging i understood what make this error happen. So basically is on line 8, you wrote `value="{{ $step['name'] }}" `  your are displaying the name of your step, when you're getting from database it works fine, but when you want to add new step field, initially it doesn't have a key called 'name' because in your increment method you are just adding a new key with type of int with a value of int also, -- see picture bellow to understand more -- and that's what cause this issue. To fix it you just have to change it to `value="{{ $step['name'] ?? '' }}"`

![problem](https://user-images.githubusercontent.com/26503233/147405062-12ae1a8d-0e8b-4d66-be9a-f92847960228.png)

**Reply by jyrjsinhatfolium** [reply]
Thanks @raidski20  It worked for me.....
